x-pack/plugins/security_solution/public/common/components/navigation/breadcrumbs/index.ts

```diff
@@ -27,33 +28,40 @@ import {
   UebaRouteSpyState,
 } from '../../../utils/route/types';
 import { getAppOverviewUrl } from '../../link_to';
-
+import { timelineActions } from '../../../../../public/timelines/store/timeline';
+import { TimelineId } from '../../../../../common/types/timeline';
 import { TabNavigationProps } from '../tab_navigation/types';
 import { getSearch } from '../helpers';
 import { GetUrlForApp, NavigateToUrl, SearchNavTab } from '../types';
 
-export const setBreadcrumbs = (
-  spyState: RouteSpyState & TabNavigationProps,
-  chrome: StartServices['chrome'],
-  getUrlForApp: GetUrlForApp,
-  navigateToUrl: NavigateToUrl
-) => {
-  const breadcrumbs = getBreadcrumbsForRoute(spyState, getUrlForApp);
-  if (breadcrumbs) {
-    chrome.setBreadcrumbs(
-      breadcrumbs.map((breadcrumb) => ({
-        ...breadcrumb,
-        ...(breadcrumb.href && !breadcrumb.onClick
-          ? {
-              onClick: (ev) => {
-                ev.preventDefault();
-                navigateToUrl(breadcrumb.href!);
-              },
-            }
-          : {}),
-      }))
-    );
-  }
+export const useSetBreadcrumbs = () => {
```

Comment:
Now that the redirect loop doesn't happen we can easily `dispatch(timelineActions.showTimeline` to close the timeline